org.eclipse.xtext/src/org/eclipse/xtext/naming/CopyQualifiedNameService.java

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.naming;
```

Comment:
put into a subpackage naming.copy
otherwise the classes would be automatically public and I'd like to keep them restricted API as before

Reply:
I doubt that we need a subpackage here. The API was stable since a couple of years already.